### ISSUE TITLE
Bug: [require-array-sort-compare] Does not work with Array.toSorted

### ISSUE THREAD

**Issue by Zamiell** [typescript-eslint/typescript-eslint#8051]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.2&fileType=.tsx&code=MYewdgzgLgBAZiEMC8MDaBGANDATDgZgF0BuAWAChKEQA6CEAJygAoBKcim2qEAZSZQApgBN2nIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aRAR1ko9oENo0foVrIA9tDplxAW2KDE6KIiGTI4MAF8QOoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const foo = [1, 2, 3];

foo.sort(); // Error, works as expected
foo.toSorted(); // No error, but should be an error
```
